src/main/java/net/fabricmc/loader/util/version/SemanticVersionPredicateParser.java

```diff
@@ -44,6 +45,11 @@
 				}
 			}
 
+			// In dev environment, ignore groovy template placeholders - any version is satisfactory
+			if (FabricLoader.getInstance().isDevelopmentEnvironment() && s.matches("(\\$\\{\\w+})")) {
```

Comment:
The regex doesn't match things like `$version`, which [also work](https://github.com/Juuxel/Adorn/blob/eed4bd355e57372ed61de24a906817210fa5fe34/src/main/resources/fabric.mod.json#L4) with Gradle.